challenge_1/x86/x95/src/reversestring.s

```diff
@@ -0,0 +1,90 @@
+.text
+.global _start
+_start:
+	#example call to function
+    pushq $requestinput
+    pushq $requestbuffer
+    call reverse_string
+    #example output
+    movq $1,%rax
+    movq $1,%rdi
+	movq $requestbuffer,%rsi
+	movq $5,%rdx
+	syscall
+	#Exit(0)
+    movq     $60,%rax
+    xorq     %rdi,%rdi
+    syscall
+
+#Function for reversing an input string. Stack contains inputstring as well as a buffer. The caller needs to provide the buffer.
+reverse_string:
+	#x86 Function Prolog
+	pushq %rbp
+	movq %rsp, %rbp
+	#save registers before smearing them
+	pushq %rsi
+	pushq %rdi
```

Comment:
The order of saving registers doesn't matter. It is just to preserve the registers of the caller (if it used them). However there is an error in the epilog which I will fix.